### PR TITLE
api: support AccessLog filter

### PR DESCRIPTION
**Pull request by zirain**
xref: https://github.com/envoyproxy/gateway/issues/3667

if we plan to only support CEL filter, the API could be more simply.

### CODE REVIEW

File: api/v1alpha1/accesslogging_types.go [hunk 1]
```diff
@@ -19,12 +19,39 @@ type ProxyAccessLog struct {
 type ProxyAccessLogSetting struct {
 	// Format defines the format of accesslog.
 	Format ProxyAccessLogFormat `json:"format"`
+	// Filters defines the extensions filters of accesslog.
+	// +notImplementedHide
+	Filters []ProxyAccessLogFilter `json:"filters,omitempty"`
```
Comment (arkodg) [inline api/v1alpha1/accesslogging_types.go:24]: thoughts on ?
```
matches:
- "response.code >= 400"
- "(connection.mtls && request.headers['x-log-mtls'] == 'true') || request.url_path.contains('v1beta3')"
```

Comment (zirain) [reply api/v1alpha1/accesslogging_types.go:24]: if we plan to support CEL only, it's fine.

Comment (arkodg) [reply api/v1alpha1/accesslogging_types.go:24]: looks like I was able to convince @guydc https://github.com/envoyproxy/gateway/issues/3667#issuecomment-2187363606 :) 

also should the noun be `filters` or `matches`

File: api/v1alpha1/accesslogging_types.go [hunk 2]
```diff
@@ -19,6 +19,10 @@ type ProxyAccessLog struct {
 type ProxyAccessLogSetting struct {
 	// Format defines the format of accesslog.
 	Format ProxyAccessLogFormat `json:"format"`
+	// Matches defines the match conditions for accesslog in CEL expression.
+	// Invalid expressions will be ignored.
```
Comment (arkodg) [inline api/v1alpha1/accesslogging_types.go:23]: lets add a link in here to highlight all the available options

File: api/v1alpha1/accesslogging_types.go [hunk 3]
```diff
@@ -19,6 +19,10 @@ type ProxyAccessLog struct {
 type ProxyAccessLogSetting struct {
 	// Format defines the format of accesslog.
 	Format ProxyAccessLogFormat `json:"format"`
+	// Matches defines the match conditions for accesslog in CEL expression.
```
Comment (arkodg) [inline api/v1alpha1/accesslogging_types.go:22]:
```suggestion
	// Matches defines the match conditions for accesslog in CEL expression.
	// An accesslog will be emitted only when one or more match conditions are evaluated to true.
```

File: api/v1alpha1/accesslogging_types.go [hunk 4]
```diff
@@ -19,6 +19,10 @@ type ProxyAccessLog struct {
...
+	// Invalid [CEL](https://www.envoyproxy.io/docs/envoy/latest/xds/type/v3/cel.proto.html#common-expression-language-cel-proto) expressions will be ignored.
```
Comment (arkodg) [inline api/v1alpha1/accesslogging_types.go:23]: can we link to a page that shows the CEL examples ? along with all available attributes

Comment (zirain) [reply api/v1alpha1/accesslogging_types.go:24]: it's on that page
image

File: api/v1alpha1/accesslogging_types.go [hunk 5]
```diff
@@ -20,6 +20,7 @@ type ProxyAccessLogSetting struct {
 	// Format defines the format of accesslog.
 	Format ProxyAccessLogFormat `json:"format"`
 	// Matches defines the match conditions for accesslog in CEL expression.
+	// An accesslog will be emitted only when one or more match conditions are evaluated to true.
```
Comment (arkodg) [inline api/v1alpha1/accesslogging_types.go:23]: curious if we can use CEL to validate the CEL 🙃  ? Or will it be hard keeping track of upstream OSS attributes?

**Review by arkodg — APPROVED** [review]
LGTM thanks !

File: api/v1alpha1/accesslogging_types.go (hunk 5 above)
Comment (zirain) [reply api/v1alpha1/accesslogging_types.go:23]: TBH, I don't know, mostly can't.

**Review by zhaohuabing — APPROVED** [review]
🚀

**Pull request merged**